keyboards/novelpad/keymaps/default/keymap.c

```diff
@@ -14,111 +14,60 @@ GNU General Public License for more details.
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
+
 #include QMK_KEYBOARD_H
 
 enum custom_keycodes {
-  BL = SAFE_RANGE,
-  WK_RED,
-  WK_GREEN,
-  WK_BLUE
+    WK_RED = SAFE_RANGE,
+    WK_GREEN,
+    WK_BLUE
 };
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-
-  [0] = LAYOUT(
-    KC_NLCK, KC_PSLS, KC_PAST, KC_ESC,
-    KC_P7,   KC_P8,   KC_P9,   KC_PMNS,
-    KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
-    KC_P1,   KC_P2,   KC_P3,   KC_TAB,
-    MO(1),   KC_P0,   KC_PDOT, KC_ENT
-  ),
-
-  [1] = LAYOUT(
-    _______, BL,      RGB_MODE_SWIRL, RESET,
-    RGB_TOG, RGB_MOD, RGB_MODE_PLAIN, RGB_MODE_SNAKE,
-    RGB_HUI, RGB_SAI, RGB_VAI,        RGB_MODE_KNIGHT,
-    RGB_HUD, RGB_SAD, RGB_VAD,        RGB_MODE_XMAS,
-    _______, WK_RED,  WK_GREEN,       WK_BLUE
-  ),
-
+    [0] = LAYOUT(
+        KC_NLCK, KC_PSLS, KC_PAST, KC_ESC,
+        KC_P7,   KC_P8,   KC_P9,   KC_PMNS,
+        KC_P4,   KC_P5,   KC_P6,   KC_PPLS,
+        KC_P1,   KC_P2,   KC_P3,   KC_TAB,
+        MO(1),   KC_P0,   KC_PDOT, KC_ENT
+    ),
+
+    [1] = LAYOUT(
+        _______, BL_STEP, RGB_M_SW, RESET,
+        RGB_TOG, RGB_MOD, RGB_M_P,  RGB_M_SN,
+        RGB_HUI, RGB_SAI, RGB_VAI,  RGB_M_K,
+        RGB_HUD, RGB_SAD, RGB_VAD,  RGB_M_X,
+        _______, WK_RED,  WK_GREEN, WK_BLUE
+    )
 };
 
 void matrix_init_user(void) {
```

Comment:
My understanding is `matrix_init_user` runs too early for this to work, and why it often seen in `matrix_scan_user`
```suggestion
void keyboard_post_init_user(void) {
```

Reply:
Good point. I almost thought it did work through the magic of EEPROM, but `rgblight_setrgb()` doesn't write to it, and has no `_noeeprom` version.